src/protos/grpc/master_metadata_service.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto3";
+
+package protos.grpc;
+
+import "google/protobuf/empty.proto";
+import "src/protos/metadata.proto";
+
+service MasterMetadataService {
+  // Create a new file chunk for |filename|, and return the associated
+  // chunk index and metadata. If |filename| already exists, it creates and
```

Comment:
If you need to create a file and write two blocks of data. How do you do this?

Option 1: `Create` has to pass a `chunk_index`, and master allocates the chunk handle for the chunk index. So, what if client tries to write a chunk index that doesn't exist, or skip a chunk index (so only create index 1 and 3, not 2?)

Option 2: `Create` means creating a new chunk, which simplifies all these things

Reply:
Well, I think create and write two blocks of data are two separate steps. First you create a file, which does not even have to create a chunk handle (and no chunk index). The only thing it causes underneath is to let master create a lock for this file and assign a default (empty) file meta data, i.e. metadata initialization.

Then you call write, which instructs the master to allocate chunks for client and client write the data to chunk server. 

I think our design support a client to write at arbitrary chunk index, because we have a map between chunk index to chunk handle for each file, so the underlying chunk index does not have to be contiguous. (Though we don't have to support such nuanced case). 

This is why I think Create and open does not need chunk_index, because no resource is allocated for this type of action.